python/ql/src/Variables/LoopVariableCapture/LoopVariableCaptureQuery.qll

```diff
@@ -59,7 +59,7 @@ module EscapingCaptureFlowConfig implements DataFlow::ConfigSig {
   predicate allowImplicitRead(DataFlow::Node node, DataFlow::ContentSet cs) {
     isSink(node) and
     (
-      cs instanceof DataFlow::TupleElementContent or
+      cs.(DataFlow::TupleElementContent).getIndex() in [0 .. 10] or
```

Comment:
Another way to do this, which doesn't restrict the index, is to properly implement `ContentSet`, and then have a branch of `ContentSet` represent all possible `TupleElementContent`s, like e.g. [Ruby has](https://github.com/github/codeql/blob/c11ed6d1190c0a8212d4b70a51801180f2debee8/ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll#L658) a branch that represent all possible element contents.

Reply:
Indeed; but it seems like it would be quite a significant task to properly implement `ContentSet` for python.